### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by lukejacksonn**
Handles no children return with empty array instead of empty string, adds tests with [tead](https://github.com/teadjs/tead). Any tips or ideas for tests?

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,7 +4,13 @@
...
   "module": "index.js",
+  "scripts": {
+    "test": "tead"
+  },
   "keywords": ["vdom", "html", "hyperscript", "hyperapp", "picodom", "preact"],
   "author": "Luke Jackson",
-  "license": "MIT"
+  "license": "MIT",
+  "dependencies": {
+    "tead": "^0.4.3"
```
Comment (Mytrill) [inline package.json]: Isn't tead a devDependency?

File: package.json [hunk 2]
```diff
@@ -4,7 +4,13 @@
   "description": "Transforms arrays into virtual dom trees",
   "main": "index.js",
   "module": "index.js",
+  "scripts": {
+    "test": "tead"
```
Comment (Mytrill) [inline package.json]: Why not adding code coverage as well :`tead --coverage` ?

File: package.json (hunk 1 above)
Comment (okwolf) [reply package.json]: Agree.

Comment (lukejacksonn) [reply package.json]: Ooops. Thanks!

File: package.json (hunk 2 above)
Comment (lukejacksonn) [reply package.json]: Will do!

Comment (lukejacksonn) [reply package.json]: Wait.. that wasn't as simple as it sounded 😅 
```
$ tead --coverage
/bin/sh: npx: command not found
Done in 0.22s.
```

Comment (okwolf) [reply package.json]: Are you using Node.js < 8?

See the note here: https://github.com/teadjs/tead#npx

Comment (lukejacksonn) [reply package.json]: I'm on `v8.1.3` but I am not using npx anywhere in the project, or at all in fact

Comment (okwolf) [reply package.json]: Sorry, forgot that`npx` wasn't introduced in Node.js 8.0.0, but comes bundled with [`npm` >= 5.2.0](https://github.com/npm/npm/releases/tag/v5.2.0), which comes bundled with [Node.js >= 8.2.0](https://github.com/nodejs/node/pull/14235). 😂 

It's not like I've [already posted about this](https://github.com/hyperapp/hyperapp/issues/518#issuecomment-355816185) and forgot or anything 🙄 

`npx` is a pretty cool tool that allows you to run a command from any package on npm without having to add it as a dependency, but the user still has control over which version they run.   First it checks if the package is installed in `node_modules`, then globally, and finally it just downloads a copy of the latest version on-demand from npm as a last resort. 😎

Comment (okwolf) [reply package.json]: FYI:
screen shot 2018-01-27 at 13 32 04
screen shot 2018-01-27 at 13 32 17

**Comment by lukejacksonn** [comment]
Going to merge this now. Much improvement on no tests 😅 thanks @Mytrill and @okwolf

**Comment by okwolf** [comment]
@lukejacksonn depending on how much more development/refactoring you're going to do on the project, you might also want a script that calls `tead --watch` like `npm run watch` or `npm run test:watch`.

**Comment by lukejacksonn** [comment]
@okwolf ahh yeh.. thanks!

**Pull request merged**